linode/manager
# feat: [M3-7686] - Restricted access UX for Databases

## Description 📝
To prevent unauthorized access to specific flows and provide clearer guidance, we aim to restrict entry to users without the required permissions.

The views under /databases have been updated to disable controls for restricted users


## Changes  🔄
- "Create Database Cluster" button on the landing page (with or without entries) is disabled for users without "add_databases" access
- All buttons & inputs in the Summary, Backups, Resize, Settings tabs under Database details have been disabled for users with "read_only" access
- Added a notification for restricted access users on the database details page

## Target release date 🗓️
NA

## Preview 📷

| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/user-attachments/assets/e32b78ad-9a39-4241-946a-58c62674f177" /> | <video src="https://github.com/user-attachments/assets/32d40bce-a1e9-4b07-bb44-b47a7d2a334a" /> |
|![Screenshot 2024-08-20 at 10 12 41 AM](https://github.com/user-attachments/assets/5c2be9d6-74bb-4665-97b0-ba9bffa8e3ec) | ![Screenshot 2024-08-20 at 10 13 07 AM](https://github.com/user-attachments/assets/05e522fb-8af2-480d-a0bc-200dc0d25120) |




## How to test 🧪

### Verification steps
1. Login as a user with full access and add databases. Grant read_only permissions to these new resources to the restricted user 
2. Login as the restricted user and navigate to the databases page. The "Create Database Cluster" button should be disabled
3. Navigate to the details page for any database - You should see a notification. Besides that, all form inputs, buttons should be disabled.
4. Login again as the user with unrestricted access and remove all access to all the databases for the restricted user. Login as the restricted user - the empty landing page for Databases should have the create button disabled.



## As an Author I have considered 🤔

*Check all that apply*
